### ISSUE TITLE
feat: Ability to modify html before PDF generation

### ISSUE THREAD

**Issue by estan** [zhaoterryy/mkdocs-pdf-export-plugin#35]
It would be great if there was a function in the theme handler in which I could modify the HTML before it's converted to PDF, for example to add a title page.

(At the moment, I tried to add a title page by combining the output with a pre-made PDF for the title page using Ghostscript, but the problem is that Ghostscript ruins all the named destinations links, so internal links won't work. Would be better if I could just insert some HTML for the title page and style it with CSS)

**Comment by zhaoterryy** [comment]
That would be 💯 great. Going to give the same reply from #29:

> Shouldn't be hard to fix. If you need this urgently I would suggest making a PR - otherwise you'll have to wait a bit.

**Comment by sander76** [comment]
You will get far doing a dedicated css regarding print styling. That's how I am planning to do this.

**Comment by estan** [comment]
Yep. I would like the content (eg title, author, version etc) that should go on the title page to be in HTML though.

**Comment by sander76** [comment]
But then you don't need to change html

**Comment by estan** [comment]
I know, but I want it in HTML because it makes sense semantically for it to be there (since it is content). And I want to take the title from the `site_name` in `mkdocs.yml`, so I was thinking that the new callback could have access to that.

Anyway, I may go with your approach in the meantime.

**Comment by sebastienlevert** [comment]
I would really like to help/comment on this issue. Is there anything in progress right now? How can we help on building this? We are looking at adding a TOC and a Cover page to our generated output and this definitely seems the way to go! Thanks!
